libs/native/libraries/build.gradle

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import org.elasticsearch.gradle.transform.UnzipTransform
+
+configurations {
+  libs {
+    attributes.attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.DIRECTORY_TYPE)
+  }
+  runtimePath
+}
+
+var zstdVersion = "1.5.5"
+
+dependencies {
+  registerTransform(UnzipTransform, transformSpec -> {
+    transformSpec.getFrom().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.JAR_TYPE);
+    transformSpec.getTo().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.DIRECTORY_TYPE);
+  });
+  libs "org.elasticsearch:zstd:${zstdVersion}:darwin-aarch64"
+  libs "org.elasticsearch:zstd:${zstdVersion}:darwin-x86-64"
+  libs "org.elasticsearch:zstd:${zstdVersion}:linux-aarch64"
+  libs "org.elasticsearch:zstd:${zstdVersion}:linux-x86-64"
+  libs "org.elasticsearch:zstd:${zstdVersion}:windows-x86-64"
+}
+
+def extractLibs = tasks.register('extractLibs', Copy)
+extractLibs.configure {
+  from configurations.libs
+  into layout.buildDirectory.dir('platform')
+  // TODO: fix architecture in uploaded libs
```

Comment:
What is referring to these? Is it just `SystemJvmOptions.findLibraryPath()`? If so, couldn't we just adapt that?

Reply:
Since we are producing the jar, we should structure the platforms with the same names we already use in the build.

Reply:
The jars are packaged such that they align with the conventions used by `Native.load()`. We can change them, but it seems to me it would make more sense for the build to align with existing Java conventions for platform naming.

Reply:
I don't know what Native.load is doing, but that is jna specific, it's not any java conventions. System.loadLibrary only looks directly on java.library.path directly, it doesn't do any recursion, and it's not looking for any platform named dirs. Since we put a platform specific dir on the library path (which is both jna and java library path), I think we should make constructing that path as easy as possible.

Reply:
Got it. We can restructure this at the same time we bundle all the native binaries into a single artifact.